lib/pure/collections/intsets.nim

```diff
@@ -46,30 +46,43 @@ type
   IntSet* = object       ## An efficient set of `int` implemented as a sparse bit set.
     elems: int           # only valid for small numbers
     counter, max: int
+    countDeleted: int
     head: PTrunk
     data: TrunkSeq
     a: array[0..33, int] # profiling shows that 34 elements are enough
 
 proc mustRehash(length, counter: int): bool {.inline.} =
+  # FACTOR with hashcommon.mustRehash
   assert(length > counter)
   result = (length * 2 < counter * 3) or (length - counter < 4)
 
-proc nextTry(h, maxHash: Hash): Hash {.inline.} =
-  result = ((5 * h) + 1) and maxHash
+proc mustRehash2[T](t: T): bool {.inline.} =
+  # FACTOR with hashcommon.mustRehash2
+  let counter2 = t.counter + t.countDeleted
+  result = mustRehash(t.max + 1, counter2)
```

Comment:
Why can't mustRehash be replaced with this?

Reply:
done (led to a few complications which is why I wanted to defer this to subsequent PRs, but I've done it now)

Reply:
Thanks!